swagger_spec_compatibility/rules/__init__.py

```diff
@@ -28,9 +45,20 @@ def compatibility_status(
     if isinstance(rules, _ALL_RULES):
         rules = RuleRegistry.rules()
 
+    rules_list = list(rules)
+
+    args_list = [
+        (old_spec, new_spec, rule)
+        for rule in rules_list
+    ]
+
+    pool = Pool(processes=4)
```

Comment:
not sure if processes=4 is good here

Reply:
4 is probably good enough. There's 9 pre-commit hooks, one being very short, the rest taking around 20-30 seconds for the big biz app spec on our hosts. 4 processes will make it take from ~3.5 minutes to ~50 seconds instead
If we set it to 8, then it'd be ~25 seconds, but 8 might be a bit much

Reply:
I'm no longer aware of the internal setup, but evaluating a single rule taking 50+ seconds seems very high.
is there a way to build up specs (such that internal API details are not published) which is able to replicate the high times observed? I think it could be a good optimisation opportunity (still having multi processing helps).